bin/dry-run.rb

```diff
@@ -476,8 +475,20 @@ def handle_dependabot_error(error:, dependency:)
 )
 
 fetcher = Dependabot::FileFetchers.for_package_manager($package_manager).new(**fetcher_args)
-$files = if $options[:clone] || always_clone
+$files = if $repo_contents_path
+           if $options[:cache_steps].include?("files") && Dir.exist?($repo_contents_path)
+            puts "=> reading cloned repo from #{$repo_contents_path}"
+           else
+            puts "=> cloning into #{$repo_contents_path}"
+            FileUtils.rm_rf($repo_contents_path)
+           end
            fetcher.clone_repo_contents
```

Comment:
FYI: This method short circuits if the repo exists https://github.com/dependabot/dependabot-core/blob/40baefd1e812feca9712daa1e359583a29b5fdc1/common/lib/dependabot/file_fetchers/base.rb#L477